docs/app-integration-development/08-OAuth-2-Functionality.md

```diff
@@ -51,5 +51,186 @@ There two options for implementing PagerDuty OAuth in your app. [PKCE (Proof Key
+ `auth_time`     |  Time when authentication occurred, in a Unix timestamp
+ `iat`           |  Time when the ID Token was issued, in a Unix timestamp
+ `purpose`       |  Purpose of the JWT. For PagerDuty ID tokens, this is always identification.
+ `at_hash`       |  ID Token hash value
+ `acr`           |  The authentication context class reference, which gives context about how the user authenticated the oauth client.
+ `delegation_id` |  ID Token delegation that tracks permissions of OAuth clients, and the history of access tokens and refresh tokens used.
+ `account_id`    |  PagerDuty account ID linked to the token
+ `user_id`       |  PagerDuty user ID linked to the token 
+ `azp`           |  The authorized party that was issued the ID Token, which is the client ID
+ `amr`           |  Authentication method references. Similar to the `acr` field, this gives context about how the user authenticated the oauth client.
+ `subdomain`     |  PagerDuty subdomain linked to the ID token 
+ `region`        |  User and account service region
+ `sid`           |  Session ID
+
+Here is a sample decrypted payload for a PagerDuty ID token:
+
+```json
+{
+  "exp": 1646086260,
+  "nbf": 1646082660,
+  "jti": "a9fd43a4-b037-4eeb-98b0-40542e3b98ff",
+  "iss": "https://app.pagerduty.com/global/oauth/anonymous",
+  "aud": [
+    "https://api.pagerduty.com",
+    "90b153bf-cfeb-4837-8b6a-864b7ff85656"
+  ],
+  "sub": "example@pagerduty.com",
+  "auth_time": 1646082649,
+  "iat": 1646082660,
+  "purpose": "id",
+  "at_hash": "LMtFKMLYeFeziNtx9PJWHA",
+  "acr": "acr:html-form:unitedstates",
+  "delegation_id": "6566027d-7707-428f-8752-3ef971b78d1a",
+  "account_id": "P9JEHMK",
+  "user_id": "P8OGMIK",
+  "azp": "90b153bf-cfeb-4837-8b6a-864b7ff85656",
+  "amr": "acr:html-form:unitedstates",
+  "subdomain": "pdt-sample",
+  "region": "UnitedStates",
+  "sid": "Ldu2m7FqnP7cH6Gl"
+}
+```
+
+### Signature
+
+The signature is used to verify that the message still has its original integrity and hasn't been tampered with and is sent from a trusted issuer.
+
+You can find the public key that PagerDuty uses to sign the token at the following URL: https://app.pagerduty.com/global/oauth/anonymous/jwks Please note however that the signing key could be rotated at any time, so we would recommend fetching the signing key from PagerDuty on each use instead of storing it locally.
+
+This signature is created using the hash algorithm from the header, and to verify a token, you must do the following:
+
+* Retrieve the algorithm used to hash the token from the header
+* Use the `x5t` or `kid` parameter from the header to retrieve the public key
+* Seperate the signature from the message
+* Convert the remaining parts to an ASCII array
+* Decode the signature using Base64Url
+* Use the decoded signature to validate the ASCII array generated previously
+
+## Decoding the full ID token with Node and Javascript
```

Comment:
Should we add an example of what a decoded id token payload looks like?

Reply:
There is one above

Reply:
ahh yes I missed that my bad 😅